src/ripple/overlay/impl/Handshake.cpp

```diff
@@ -239,8 +239,7 @@ verifyHandshake(
             throw std::runtime_error("Invalid server domain");
     }
 
-    // Check the network. Omitting Network-ID (on either side ours, or theirs)
-    // means NID=0
+    if (auto const iter = headers.find("Network-ID"); iter != headers.end())
```

Comment:
This doesn't seem right. Its synced code but it might have been merged correctly. These 2 calls are redundant